### PR TITLE
P4C-DPDK - Support Non-zero arguments for default action

### PR DESCRIPTION
**Pull request by github-sajan**

### CODE REVIEW

File: backends/dpdk/dpdkArch.cpp [hunk 1]
```diff
@@ -1255,6 +1255,22 @@ const IR::Node *CollectLocalVariables::postorder(IR::P4Parser *p) {
...
+void DefActionValue::postorder(const IR::P4Table *t) {
+    auto default_action = t->properties->getProperty("default_action");
+    if (default_action != nullptr && default_action->value->is<IR::ExpressionValue>()) {
+        auto expr = default_action->value->to<IR::ExpressionValue>()->expression;
+        auto mi = P4::MethodInstance::resolve(expr->to<IR::MethodCallExpression>(),
+                refMap, typeMap);
+        BUG_CHECK(mi->is<P4::ActionCall>(),
+            "%1%: expected action in default_action", default_action);
+        auto new_l = new IR::IndexedVector<IR::Parameter>;
+        for (auto p : mi->to<P4::ActionCall>()->action->parameters->parameters) {
```
Comment (mbudiu-vmw) [inline backends/dpdk/dpdkArch.cpp:1267]: why can't you just keep a pointer to the list itself?

File: backends/dpdk/dpdkArch.h [hunk 2]
```diff
@@ -515,6 +515,19 @@ class PrependPDotToActionArgs : public Transform {
     const IR::Node *preorder(IR::MethodCallExpression*) override;
 };
 
+class DefActionValue : public Inspector {
```
Comment (mbudiu-vmw) [inline backends/dpdk/dpdkArch.h:518]: please document this class.

File: backends/dpdk/spec.cpp [hunk 3]
```diff
@@ -343,8 +343,24 @@ std::ostream &IR::DpdkTable::toSpec(std::ostream &out) const {
...
         out << " args none ";
     } else {
-        BUG("non-zero default action arguments not supported yet");
+        out << " args ";
+        auto mce = default_action->to<IR::MethodCallExpression>();
+        auto earg = mce->arguments->at(0)->expression;
+        if (earg->is<IR::ListExpression>()) {
+            auto paramCount = earg->to<IR::ListExpression>()->components.size();
+            for (unsigned i = 0; i < paramCount; i++) {
+                if (earg->to<IR::ListExpression>()->components.at(i)->is<IR::Constant>()) {
```
Comment (mbudiu-vmw) [inline backends/dpdk/spec.cpp:352]: so if the argument is not a numeric constant what should happen?

**Review by mbudiu-vmw — CHANGES_REQUESTED** [review]

File: backends/dpdk/dpdkArch.h (hunk 2 above)
Comment (github-sajan) [reply backends/dpdk/dpdkArch.h:526]: Added comments

File: backends/dpdk/spec.cpp (hunk 3 above)
Comment (github-sajan) [reply backends/dpdk/spec.cpp:353]: Addressed

**Review by mbudiu-vmw — APPROVED** [review]

**Pull request merged**